c/rce.h

```diff
@@ -0,0 +1,382 @@
+#ifndef XUDT_RCE_SIMULATOR_C_RCE_H_
+#define XUDT_RCE_SIMULATOR_C_RCE_H_
+#include "ckb_smt.h"
```

Comment:
SMT C implementation should be well tested, I'm especially not confident with `smt_state_normalize()` function.

Reply:
I've checked the version used here with Godwoken's: they are same (except naming).